pkg/lorry/plugin/proto/db_plugin.proto

```diff
@@ -0,0 +1,256 @@
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the new member's Pod.
+  string new_member = 5;
+
+  // The IP of the new member's Pod.
+  string new_member_ip = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message JoinMemberResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message LeaveMemberRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica to check the role.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the leave member's Pod.
+  string leave_member = 5;
+
+  // The IP of the leave member's Pod.
+  string leave_member_ip = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message LeaveMemberResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadonlyRequest {
+  DBInfo db_info = 1;
+
+  // The reason for setting the replica as read-only.
+  string Reason = 2;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 3;
+}
+
+message ReadonlyResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadwriteRequest {
+  DBInfo db_info = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message ReadwriteResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message AccountProvisionRequest {
+  DBInfo db_info = 1;
+
+  // the user name of the new account.
+  string user_name = 2;
+
+  // the password of the new account.
+  string password = 3 [(kb_secret) = true];
+
+  // the role of the new account.
+  string role = 4;
+
+  // Common metadata property for extention
+}
+
+message AccountProvisionResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message SwitchoverRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Name of the old primary replica to demote.
+  string primary = 4;
+
+  // The Name of the new primary replica.
+  string candidate = 5;
+
+  // Force the switchover to happen even if the primary is not ready.
+  bool force = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message SwitchoverResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+// DBPlugin service provides APIs to kubeblocks operator to exec component operation.
+service DBPlugin {
```

Comment:
"DBPlugin" may not be the most appropriate name, as we have add-ons like Minio that are not necessarily database-related.

Reply:
update to ServicePlugin

Reply:
Although it's much longer, ComponentLifecyclePlugin is better.

Reply:
Actions may exist at different levels, such as component level, replica level, or even cluster level. Considering this, would it be more appropriate to use "lifecycleplugin" as the name?